### PR TITLE
boomerang: switch to new active fork, cleanup, maintain

### PR DESCRIPTION
**Pull request by dtzWill**
Fixes many bugs and GUI doesn't have missing icons, etc.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/boomerang/default.nix [hunk 1]
```diff
@@ -1,38 +1,43 @@
-{ stdenv, fetchgit, cmake, expat, qt5, boost }:
+{ stdenv, fetchFromGitHub, cmake, qtbase }:
 
 stdenv.mkDerivation rec {
   name = "boomerang-${version}";
-  version = "0.3.99-alpha-2016-11-02";
+  version = "0.4.0-alpha-20180118";
```
Comment (orivej) [inline pkgs/development/tools/boomerang/default.nix]: Could you keep hyphens in the date? (https://nixos.org/nixpkgs/manual/#sec-package-naming documents that `YYYY-MM-DD` is the preferred format)

Why are you changing `0.3.99` to `0.4.0`? The latest tag is `v0.3.99-alpha`, so leaving it as is seems appropriate.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -807,7 +807,7 @@ with pkgs;
 
   borgbackup = callPackage ../tools/backup/borg { };
 
-  boomerang = callPackage ../development/tools/boomerang { };
+  boomerang = qt5.callPackage ../development/tools/boomerang { };
```
Comment (orivej) [inline pkgs/top-level/all-packages.nix]: Please use `libsForQt5.callPackage`.

Comment (dtzWill) [reply pkgs/top-level/all-packages.nix]: I somehow got it into my head that qt5.callPackage was no preferred over libsForQt5; I'm not sure where this came from, and will of course fix it :).

File: pkgs/development/tools/boomerang/default.nix (hunk 1 above)
Comment (dtzWill) [reply pkgs/development/tools/boomerang/default.nix]: It's the version the code gives itself currently: https://github.com/ceeac/boomerang/blob/dc53653f2374676c07296b4c51eed4ac4eeba127/cmake-scripts/boomerang-version.cmake#L11

Happy to conform with date, wasn't sure how to handle this.
My understanding is that `2018-01-18` (nothing else) is preferred, but didn't want to "break" upgrades by anyone currently having it installed.  Will change as you suggest.

**Pull request merged**